optimizely/client/factory.go

```diff
@@ -111,6 +116,7 @@ func (f OptimizelyFactory) ClientWithOptions(clientOptions Options) (*Optimizely
 		client.decisionService = decision.NewCompositeService(notificationCenter)
 	}
 
+	client.eventProcessor = event.NewEventProcessor(defaultEventQueueSize, defaultEventFlushInterval)
```

Comment:
- Are we going to add more configurability, like allowing passing in event processor, or event processor options?
- Should `ctx` be provided to the event processor, so that `cancelFunc` can shut it down also?

Reply:
Yes to both. I am planning to add more configurability later on as people start asking about it during beta. 

The ctx should also be passed in but that requires a small refactor of event processor that we can take up in a separate PR (I think Tom might be looking at taking in the context already as part of his work).

Regardless, I will add the TODOs here so we know to go back and add em.